### PR TITLE
feat(scripts): pwsh script for manipulating the yarn cache

### PR DESCRIPTION
**Pull request by Quantumlyy**
The current option is only for clearing it since its the most used.

### CODE REVIEW

File: scripts/yarncache.ps1 [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    Param (
+        [string]$Command = "default"
+    )
+
+    Process {
+        switch ( $Command ) {
+            clear {
+                Remove-Item -r -Force dist
+				Remove-Item -r -Force node_modules
+				Remove-Item -r -Force $(yarn cache dir)
```
Comment (kyranet) [inline scripts/yarncache.ps1]:
```suggestion
                Remove-Item -Recurse -Force dist
				Remove-Item -Recurse -Force node_modules
				Remove-Item -Recurse -Force $(yarn cache dir)
```

**Review by kyranet — DISMISSED** [review]
You should also use GetCommand to see if `yarn` exists, and run the similar commands.

File: scripts/dependencycache.ps1 [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    )
+
+    Get-Process {
+        switch ( $Command ) {
+            clear {
+                Remove-Item -Recurse -Force dist
+				Remove-Item -Recurse -Force node_modules
+				switch ($Manager) {
+					yarn {
+						Remove-Item -Recurse -Force $(yarn cache dir)
```
Comment (favna) [inline scripts/dependencycache.ps1]:
```suggestion
						Remove-Item ((yarn cache dir) + "\*") -Recurse -Force
```

We have to delete the items *within* the cache, not the cache dir itself. Taken from my snippets in Diri (yarncache Skyra tag) which are tested and I use every so often.

File: scripts/dependencycache.ps1 [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+					}
```
Comment (Favna) [inline scripts/dependencycache.ps1:15]:
```suggestion
						Remove-Item -Force -ErrorAction Ignore yarn.lock
						yarn install
					}
```

Also remove the lockfile which holds the Git SHA then run yarn install

File: scripts/dependencycache.ps1 [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+					npm {
+						npm cache rm
```
Comment (favna) [inline scripts/dependencycache.ps1]:
```suggestion
						npm cache rm
						Remove-Item -Force -ErrorAction Ignore package-lock.json
						Remove-Item -Force -ErrorAction Ignore npm-shrinkwrap.json
						npm install
```

Same as before but for NPM

File: scripts/dependencycache.ps1 [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    Param (
+		[string]$Command = "default",
+		[string]$Manager = "yarn"
+    )
+
+    Get-Process {
+        switch ( $Command ) {
+            clear {
+                Remove-Item -Recurse -Force dist
+				Remove-Item -Recurse -Force node_modules
```
Comment (favna) [inline scripts/dependencycache.ps1]:
```suggestion
                Remove-Item -Recurse -Force -ErrorAction Ignore dist
				Remove-Item -Recurse -Force -ErrorAction Ignore node_modules
```

Script should not exist if the folders do not exist

**Review by favna — CHANGES_REQUESTED** [review]

File: scripts/dependencycache.ps1 (hunk 2 above)
Comment (kyranet) [reply scripts/dependencycache.ps1]: Doesn't matter, Yarn re-creates the directory.

File: scripts/dependencycache.ps1 (hunk 3 above)
Comment (kyranet) [inline scripts/dependencycache.ps1:15]: That'd force the update and end up updating some hashes unintentionally. `yarn upgrade-interactive --latest` should be used instead.

**Review by favna — APPROVED** [review]

**Review by favna — DISMISSED** [review]

**Review by favna — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**